fontscan/substitutions.go

```diff
@@ -0,0 +1,254 @@
+package fontscan
+
+import (
+	"container/list"
+	"strings"
+
+	"github.com/go-text/typesetting/language"
+	meta "github.com/go-text/typesetting/opentype/api/metadata"
+)
+
+// this file implements the family substitution feature,
+// inspired by fontconfig.
+// it works by defining a set of modifications to apply
+// to a user provided family
+// each of them may happen one (or more) alternative family to
+// look for
+
+// familySubstitution maps family name to possible alias
+// it is generated from fontconfig substitution rules
+// the order matters, since the rules apply sequentially to the current
+// state of the family list
+func init() {
+	// replace families keys by their no case no blank version
+	for i, v := range familySubstitution {
+		for i, s := range v.additionalFamilies {
+			v.additionalFamilies[i] = meta.NormalizeFamily(s)
+		}
+
+		familySubstitution[i].test = v.test.normalize()
+	}
+}
+
+// we want to easily insert at the start,
+// the end and "around" an element
+type familyList struct {
+	*list.List
+}
+
+// panic if `families` is empty
```

Comment:
This doesn't look like it actually will panic. Old docs?

Reply:
Indeed, thanks !